sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,232 @@
+from sympy.core.symbol import symbols
+from sympy.physics.mechanics import dynamicsymbols, Body, PinJoint
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.vector import Vector, ReferenceFrame
+from sympy.testing.pytest import raises
+
+t = dynamicsymbols._t
+
+
+def test_Joint():
+    parent = Body('parent')
+    child = Body('child')
+    raises(TypeError, lambda: Joint('J', parent, child))
+
+
+def test_pinjoint():
+    P = Body('P')
+    C = Body('C')
+    l, m = symbols('l m')
+    theta, omega = dynamicsymbols('theta_J, omega_J', real=True)
+
+    Pj = PinJoint('J', P, C )
+    assert Pj._name == 'J'
+    assert Pj.parent() == P
+    assert Pj.child() == C
+    assert Pj.coordinates() == [theta]
+    assert Pj.speeds() == [omega]
+    assert Pj.kdes() == [omega - theta.diff(t)]
+    assert Pj._parent_axis == P.frame.x
+    assert Pj._child_joint.pos_from(C.masscenter) == Vector(0)
+    assert Pj._parent_joint.pos_from(P.masscenter) == Vector(0)
+    assert Pj._parent_joint.pos_from(Pj._child_joint) == Vector(0)
+    assert C.masscenter.pos_from(P.masscenter) == Vector(0)
+
+    P1 = Body('P1')
+    C1 = Body('C1')
+    J1 = PinJoint('J1', P1, C1, parent_joint_pos= l * P1.frame.x, child_joint_pos= m * C1.frame.y,
+        parent_axis = P1.frame.z)
+
+    assert J1._parent_axis == P1.frame.z
+    assert J1._child_joint.pos_from(C1.masscenter) == m * C1.frame.y
+    assert J1._parent_joint.pos_from(P1.masscenter) == l * P1.frame.x
+    assert J1._parent_joint.pos_from(J1._child_joint) == Vector(0)
+    assert P1.masscenter.pos_from(C1.masscenter) == - l*P1.frame.x + m*C1.frame.y
+
+def test_pin_joint_double_pendulum():
+    q1, q2 = dynamicsymbols('q1 q2')
+    u1, u2 = dynamicsymbols('u1 u2')
+    m, l = symbols('m l')
+    N = ReferenceFrame('N')
+    A = ReferenceFrame('A')
+    B = ReferenceFrame('B')
+    C = Body('C', frame=N) #ceiling
+    PartP = Body('P', frame=A, mass=m)
+    PartR = Body('R', frame=B, mass=m)
+
+    J1 = PinJoint('J1', C, PartP, speeds=u1, coordinates=q1, child_joint_pos=-l*A.x, \
+        parent_axis=C.frame.z, child_axis=PartP.frame.z)
+    J2 = PinJoint('J2', PartP, PartR, speeds=u2, coordinates=q2, child_joint_pos=-l*B.x, \
+        parent_axis=PartP.frame.z, child_axis=PartR.frame.z)
+
+    #Check orientation
+    assert N.dcm(A) == Matrix([[cos(q1), -sin(q1), 0], [sin(q1), cos(q1), 0], [0, 0, 1]])
+    assert A.dcm(B) == Matrix([[cos(q2), -sin(q2), 0], [sin(q2), cos(q2), 0], [0, 0, 1]])
+    assert N.dcm(B).simplify() == Matrix([[cos(q1 + q2), -sin(q1 + q2), 0], \
+        [sin(q1 + q2), cos(q1 + q2), 0], [0, 0, 1]])
+
+    #Check Angular Velocity
+    assert A.ang_vel_in(N) == u1 * N.z
+    assert B.ang_vel_in(A) == u2 * A.z
+    assert B.ang_vel_in(N) == u1 * N.z + u2 * A.z
+
+    #Check kde
+    assert J1.kdes() == [u1 - q1.diff(t)]
+    assert J2.kdes() == [u2 - q2.diff(t)]
+
+    #Check Linear Velocity
+    assert PartP.masscenter.vel(N) == l*u1*A.y
```

Comment:
You don't have a test to show that the lower pendulum's velocity in the N frame is correct.